cytominer_database/ingest.py

```diff
@@ -69,48 +69,36 @@ def into(input, output, name, identifier, skip_table_prefix=False):
-        # but with the column names prefixed with the name of the compartment
-        # (or `Image`, if this is an image CSV file, and `skip_table_prefix` is False)
-        with open(input, "r") as fin, open(source, "w") as fout:
-            reader = csv.reader(fin)
-            writer = csv.writer(fout)
-
-            headers = next(reader)
-            if not skip_table_prefix:
-                headers = [__format__(name, header) for header in headers]
-
-            # The first column is `TableNumber`, which is the unique identifier for the image CSV
-            headers = ["TableNumber"] + headers
-
-            writer.writerow(headers)
-
-            [writer.writerow([identifier] + row) for row in reader]
-
-        # Now ingest the temp CSV file (with the modified column names) into the database backend
-        # the rows of the CSV file are inserted into a table with name `name`.
-        with warnings.catch_warnings():
-            # Suppress the following warning on Python 3:
-            #
-            #   /usr/local/lib/python3.6/site-packages/odo/utils.py:128: DeprecationWarning: inspect.getargspec() is
-            #     deprecated, use inspect.signature() or inspect.getfullargspec()
-            warnings.simplefilter("ignore", category=DeprecationWarning)
-
-            engine = create_engine(output)
-            con = engine.connect()
-
-            df = pd.read_csv(source, index_col=0)
-            df.to_sql(name=name, con=con, if_exists="append")
+    with warnings.catch_warnings():
+        # Suppress the following warning on Python 3:
+        #
+        #   /usr/local/lib/python3.6/site-packages/odo/utils.py:128: DeprecationWarning: inspect.getargspec() is
+        #     deprecated, use inspect.signature() or inspect.getfullargspec()
+        warnings.simplefilter("ignore", category=DeprecationWarning)
+        engine = create_engine(output)
+        con = engine.connect()
+
+        df = pd.read_csv(input)
+        #print("In into(): df['Index']") # no index prepended yet
+        #print(df['Index'])
+        # add "name"prefix to column headers
+        if not skip_table_prefix:
+            no_prefix = ["ImageNumber", "ObjectNumber"]  # exception columns
+            df.columns = [
+                i if i in no_prefix else name + "_" + i for i in df.columns
+            ]
+        # add TableNumber
+        number_of_rows, _ = df.shape
+        table_number_column = [identifier] * number_of_rows  # create additional column
+        df.insert(0, "TableNumber", table_number_column, allow_duplicates=False)
+        print("In into(): df.shape is ", df.shape)
+        df.to_sql(name=name, con=con, if_exists="append", index = False)
 
 def checksum(pathname, buffer_size=65536):
     """
     Generate a 32-bit unique identifier for a file.
-    
-    :param pathname: input file
-    :param buffer_size: buffer size   
+
+    :param buffer_size: buffer size
```

Comment:
```suggestion
    :param buffer_size: buffer size
    :param pathname: input file
```

Any reason to drop the `pathname` documentation? If so, feel free to ignore this suggestion.

Reply:
@diskontinuum